wis2box_api/plugins/process/dataset_info.py

```diff
@@ -0,0 +1,286 @@
+                        # fill dataset_info dict
+                        dataset_info[key] = {
+                            'topic': item['properties']['wmo:topicHierarchy'],
+                            'files_incoming_24hrs': 0,
+                            'files_public_24hrs': 0,
+                            'timestamp_last_incoming': None,
+                            'timestamp_last_public': None,
+                            'es_index': index,
+                            'es_status': None
+                        }
+            else:
+                LOGGER.error(f'Error getting collection list: {response.text}')
+                raise ProcessorExecuteError('Error getting collection list')
+        except Exception as err:
+            LOGGER.error(f'Error getting collection list: {err}')
+            raise ProcessorExecuteError('Error getting collection list')
+
+        # define date offset
+        now_minus_24hrs = datetime.now(timezone.utc) - timedelta(hours=24)
+        if self.minio_client is not None:
+            incoming_bucket_info = self._get_bucket_info(STORAGE_INCOMING, now_minus_24hrs) # noqa
+            public_bucket_info = self._get_bucket_info(STORAGE_PUBLIC, now_minus_24hrs) # noqa
+
+        for c_id in dataset_info:
+            topic = (dataset_info[c_id]['topic']).replace('origin/a/wis2/', '')
+            es_index = dataset_info[c_id]['es_index']
+            if c_id in incoming_bucket_info or topic in incoming_bucket_info:
+                key = c_id if c_id in incoming_bucket_info else topic
+                dataset_info[c_id]['files_incoming_24hrs'] = incoming_bucket_info[key]['files_last24hrs'] # noqa
+                dataset_info[c_id]['timestamp_last_incoming'] = incoming_bucket_info[key]['last_timestamp'] # noqa
+            if c_id in public_bucket_info or topic in public_bucket_info:
+                key = c_id if c_id in public_bucket_info else topic
+                dataset_info[c_id]['files_public_24hrs'] = public_bucket_info[key]['files_last24hrs'] # noqa
+                dataset_info[c_id]['timestamp_last_public'] = public_bucket_info[key]['last_timestamp'] # noqa
+            if dataset_info[c_id]['timestamp_last_incoming'] is not None:
+                dataset_info[c_id]['timestamp_last_incoming'] = dataset_info[c_id]['timestamp_last_incoming'].astimezone(timezone.utc).strftime('%Y-%m-%dT%H:%M:%SZ') # noqa
+            if dataset_info[c_id]['timestamp_last_public'] is not None:
+                dataset_info[c_id]['timestamp_last_public'] = dataset_info[c_id]['timestamp_last_public'].astimezone(timezone.utc).strftime('%Y-%m-%dT%H:%M:%SZ') # noqa
+            if self.es is not None and es_index != 'notfound':
+                dataset_info[c_id]['es_status'] = self._get_es_index_info(es_index) # noqa
+            continue
+        outputs = {
+            'dataset_info': dataset_info
+        }
+        return mimetype, outputs
+
+    def _get_es_index_info(self, index):
```

Comment:
I'm not sure we should be emitting settings information without access control?  Could we make the resulting process endpoint protected?  For discussion.

Reply:
I am not sure how we can make the resulting process endpoint protected without incurring additional work in the other repositories.